browser/src/shared/tracking/eventLogger.tsx

```diff
@@ -0,0 +1,89 @@
+import uuid from 'uuid'
+import * as GQL from '../../../../shared/src/graphql/schema'
+import { TelemetryService } from '../../../../shared/src/telemetry/telemetryService'
+import { storage } from '../../browser/storage'
+import { isInPage } from '../../context'
+import { logUserEvent } from '../backend/userEvents'
+
+const uidKey = 'sourcegraphAnonymousUid'
+
+export class EventLogger implements TelemetryService {
+    private uid: string | null = null
+
+    constructor() {
+        // Fetch user ID on initial load.
+        this.getAnonUserID().then(
+            () => {
+                /* noop */
+            },
+            () => {
+                /* noop */
+            }
+        )
+    }
+
+    /**
+     * Generate a new anonymous user ID if one has not yet been set and stored.
+     */
+    private generateAnonUserID = (): string => uuid.v4()
+
+    /**
+     * Get the anonymous identifier for this user (allows site admins on a private Sourcegraph
+     * instance to see a count of unique users on a daily, weekly, and monthly basis).
+     *
+     * Not used at all for public/Sourcegraph.com usage.
+     */
+    private async getAnonUserID(): Promise<string> {
+        if (this.uid) {
+            return this.uid
+        }
+
+        if (isInPage) {
+            let id = localStorage.getItem(uidKey)
+            if (id === null) {
+                id = this.generateAnonUserID()
+                localStorage.setItem(uidKey, id)
+            }
+            this.uid = id
+            return this.uid
+        }
+
+        let { sourcegraphAnonymousUid } = await storage.sync.get()
+        if (!sourcegraphAnonymousUid || sourcegraphAnonymousUid === '') {
```

Comment:
```suggestion
        if (!sourcegraphAnonymousUid) {
```
Empty strings are falsy